editor/editor_inspector.cpp

```diff
@@ -3230,83 +3183,48 @@ void EditorInspector::update_tree() {
-							doc_info.description = F->value.properties[i].description;
-							if (is_native_class) {
-								doc_info.description = DTR(doc_info.description); // Do not translate the property description of scripts.
-							}
-
-							const Vector<String> class_enum = F->value.properties[i].enumeration.split(".");
-							const String class_name = class_enum[0];
-							const String enum_name = class_enum.size() >= 2 ? class_enum[1] : "";
-							if (!enum_name.is_empty()) {
-								HashMap<String, DocData::ClassDoc>::ConstIterator enum_class = dd->class_list.find(class_name);
-								if (enum_class) {
-									for (DocData::ConstantDoc val : enum_class->value.constants) {
-										// Don't display `_MAX` enum value descriptions, as these are never exposed in the inspector.
-										if (val.enumeration == enum_name && !val.name.ends_with("_MAX")) {
-											const String enum_value = EditorPropertyNameProcessor::get_singleton()->process_name(val.name, EditorPropertyNameProcessor::STYLE_CAPITALIZED);
-											// Prettify the enum value display, so that "<ENUM NAME>_<VALUE>" becomes "Value".
-											String desc = val.description;
-											if (is_native_class) {
-												desc = DTR(desc); // Do not translate the enum value description of scripts.
-											}
-											desc = desc.trim_prefix("\n");
-											doc_info.description += vformat(
-													"\n[b]%s:[/b] %s",
-													enum_value.trim_prefix(EditorPropertyNameProcessor::get_singleton()->process_name(enum_name, EditorPropertyNameProcessor::STYLE_CAPITALIZED) + " "),
-													desc.is_empty() ? ("[i]" + TTR("No description.") + "[/i]") : desc);
-										}
-									}
-								}
-							}
-
-							doc_info.path = "class_property:" + F->value.name + ":" + F->value.properties[i].name;
-							break;
-						}
-					}
-
+				HashMap<String, DocData::ClassDoc>::ConstIterator F = EditorHelp::get_doc_data()->class_list.find(classname);
+				if (F) {
 					Vector<String> slices = propname.operator String().split("/");
+					// Check if it's a theme item first.
 					if (slices.size() == 2 && slices[0].begins_with("theme_override_")) {
 						for (int i = 0; i < F->value.theme_properties.size(); i++) {
+							String doc_path_current = "class_theme_item:" + F->value.name + ":" + F->value.theme_properties[i].name;
 							if (F->value.theme_properties[i].name == slices[1]) {
-								doc_info.description = F->value.theme_properties[i].description;
-								if (is_native_class) {
-									doc_info.description = DTR(doc_info.description); // Do not translate the theme item description of scripts.
-								}
-								doc_info.path = "class_theme_item:" + F->value.name + ":" + F->value.theme_properties[i].name;
-								break;
+								doc_path = doc_path_current;
+								theme_item_name = F->value.theme_properties[i].name;
 							}
 						}
-					}
 
-					if (!F->value.inherits.is_empty()) {
-						F = dd->class_list.find(F->value.inherits);
```

Comment:
This is the from the loop I removed that I talked about on the very first comment (https://github.com/godotengine/godot/pull/82051#discussion_r1333178891). If you could point me to an example it would help me test this more precisely.

Reply:
Theme overrides can be one such example, because they are not displayed according to the class hierarchy and instead are displayed under the `Control`/`Window` section. So we can't rely on the category name, we need to go through each class in the hierarchy and find the description. In most other cases this shouldn't be needed though.

Notice how `CheckBox` has no option to open description with your PR:

<img width="202" alt="godot windows editor dev x86_64_2023-09-29_13-28-09" src="https://github.com/godotengine/godot/assets/11782833/71bda22f-d0ae-4f27-9c3d-f60d2328a2b4">
<img width="200" alt="godot windows editor dev x86_64_2023-09-29_13-27-59" src="https://github.com/godotengine/godot/assets/11782833/d1ed4326-2899-4a68-848f-f2e17c29f87d">

Tooltips don't show either. Actually, today for some reason I can't get a description for `Button`'s theme properties either. It worked fine for `Button` yesterday, so I can only assume there is some weirdness with the cache 🤔

It all works correctly in `master`.